tutorials/topi/intro_topi.py

```diff
@@ -0,0 +1,115 @@
+"""
+from __future__ import absolute_import, print_function
+
+import tvm
+import topi
+
+######################################################################
+# Basic example
+# -------------
+# Let's revisit the sum of rows operation (equivalent to :code:`B = numpy.sum(A, axis=1)`') \
+# from :ref:`reduction` tutorial. To compute the sum of rows of a two dimensional TVM tensor A, we should
+# specify the symbolic operation as well as schedule as follows
+#
+n = tvm.var("n")
+m = tvm.var("m")
+A = tvm.placeholder((n, m), name='A')
+k = tvm.reduce_axis((0, m), "k")
+B = tvm.compute((n,), lambda i: tvm.sum(A[i, k], axis=k), name="B")
+s = tvm.create_schedule(B.op)
+
+# and to examine the IR code in human readable format, we can do
+#
+print(tvm.lower(s, [A], simple_mode=True))
+
+# However, for such a common operation we had to define the reduce axis ourselves as well as explicit computation with
+# :code: `tvm.compute`. Imagine for more complicated operations how much details we need to provide.
+# Fortunately, we can replace those two lines with simple :code:`topi.sum` much like :code`numpy.sum`
+#
+C = topi.sum(A, axis=1)
+ts = tvm.create_schedule(C.op)
+print(tvm.lower(ts, [A], simple_mode=True))
+
+######################################################################
+# Numpy-style operator overloading
+# --------------------------------
+# We can add two tensors using :code:`topi.broadcast_add` that have correct (broadcastable with specific) shapes.
+# Even shorter, TOPI provides operator overloading for such common operations. For example,
+#
+a = tvm.placeholder((100, 10, 10), name="a")
+b = tvm.placeholder((10, 10), name="b")
+c = a + b  # same as topi.broadcast_add
+d = a * b  # same as topi.broadcast_mul
+
+# Overloaded with the same syntax, TOPI handles broadcasting a primitive (`int`, `float`) to a tensor :code:`d - 3.14`.
+
+######################################################################
+# Generic schedules and fusing operations
+# -----------------
+# Up to now, we have seen an example of how TOPI can save us from writing explicit computations in lower level API.
+# But it doesn't stop here. Still we did the scheduling as before. TOPI also provides higher level
+# scheduling recipes depending on a given context. For example, for CUDA,
+# we can schedule the following series of operations ending with :code:`topi.sum` using only
+# :code:`topi.generic.schedule_reduce`
+#
+e = topi.elemwise_sum([c, d], num_args=100 * 10 * 10)
+f = e / 2.0
+g = topi.sum(f)
+with tvm.target.cuda():
+    sg = topi.generic.schedule_reduce(g)
+    print(tvm.lower(sg, [a, b], simple_mode=True))
+
+# As you can see, scheduled stages of computation have been accumulated and we can examine them by
+#
+print(sg.stages)
+
+# As a more complicated example, we can implement _softmax_ operator using TOPI as follows
+#
+tarray = tvm.placeholder((512, 512), name="tarray")
```

Comment:
This is still here, need to be removed, because two reductions cannot be scheduled as injective